docs/observability.md

```diff
@@ -0,0 +1,149 @@
+
+In this sample we deploy a three layered network topology and want to understand the aggregated logs in Azure Monitor. To achieve this we leverage these custom processors to enrich the signals as they get exported per layer.
+Based on the spec, using the `Action` of `insert` ensures the key is added if not yet existent. When a lower level OpenTelemetry Collector moves its signals through the different levels, existing attributes will not be overwritten thus ensuring each original signal only receives the new metadata at time of first export from the initial location.
+
+Today metadata such as `edge.layer`, `edge.region` and `k8s.cluster.name` are inserted and configured at installation time by providing custom values to the Helm chart.
+There are also other options to automatically add attributes, such as Environment Variables through the [`Resource Detection Processor`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md).
+
+```yaml
+processors:
+  
+  resource:
+    attributes:
+    - key: edge.layer
+      value: {{ .Values.metadata.networkLayer }}
+      action: insert
+    - key: edge.region
+      value: {{ .Values.metadata.region }}
+      action: insert
+    - key: k8s.cluster.name
+      value: {{ .Values.metadata.clusterName }}
+      action: insert
+```
+
+Updating these values after initial Helm chart installation can be achieved by executing `helm upgrade`.
+
+Within Azure Application Insights the added attributes are available in `customDimensions` array and can be further used to query the logs, metrics and traces.
+Example of a log entry in the `traces` table in App Insights, customDimensions:
+
+```json
+{
+    "cpu":"1",
+    "edge.layer":"L4",
+    "edge.region":"northeurope",
+    "k8s.cluster.name":"[-myclusterdeploymentname-]",
+    "http.scheme":"http",
+    "instrumentationlibrary.name":"otelcol/prometheusreceiver",
+    "instrumentationlibrary.version":"0.81.0",
+    "k8s.container.name":"node-exporter",
+    "k8s.daemonset.name":"otelcollection-prometheus-node-exporter",
+    "k8s.namespace.name":"monitoring",
+    "k8s.node.name":"[...]",
+    "k8s.pod.name":"otelcollection-prometheus-node-exporter-xxx",
+    "k8s.pod.uid":"[...]",
+    "net.host.name":"[...]",
+    "net.host.port":"9100",
+    "service.instance.id":"..:9100",
+    "service.name":"node-exporter"
+}
+```
+
+### OpenTelemetry Support in Components of the Solution
+
+This section contains a number of relevant resources of (initial) built-in OpenTelemetry support for some of the technologies used in this sample.
+
+- **Envoy Proxy**:
+  - OpenTelemetry has built-in support for OpenTelemetry tracing, however it is not used in this solution since it requires TLS termination which this solution is not implementing.
+  - Prometheus compatible `/metrics` endpoint is available though exposing it through the default path requires special configuration, as is the case in this sample.
```

Comment:
maybe rephrasing this sentence...I think there is something messed up ;-)

Reply:
Wdyt? I've reworded both bullets as they were indeed not clear.